### DISCUSSION TITLE
Place current_working_dir in Tab title

### DISCUSSION THREAD

**Discussion by idr4n** [wezterm/wezterm#5890] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20240701-070926-69686f45

### Ask your question!

I want to set the tab title to the active pane's current_working_dir, as per mentioned in https://github.com/wez/wezterm/issues/915#issuecomment-1000605863

So I have this which is not workin:

```lua
wezterm.on("format-tab-title", function(tab, tabs, panes, config, hover, max_width)
  local pane = tab.active_pane
  local cwd = pane.current_working_dir
  return {
    -- { Text = " " .. " hello " .. " " }, -- this works
    { Text = " " .. cwd .. " " }, -- this doesn't work
  }
end)
```

Any suggestion on how to make this work?

Thanks in advance!

**Comment by habahabahaba** [comment]
Same question. Nothing, I tried, seems to work on MacOS.

**Comment by estin** [comment]
`pane.current_working_dir` is [Url](https://wezfurlong.org/wezterm/config/lua/wezterm.url/Url.html) object

to fix try `{ Text = " " .. cwd.path .. " " }`

**Reply by habahabahaba** [reply]
Thank you for the quick reply, Estin. ChatGPT just came to the same conclusions. 
Also `pane.current_working_dir.path` will have escaped spaces (`%20`) and possibly other characters; here's the decoder function: 
```
local function decodeURI(path)
  if not path then
    return ''
  end
  -- Replace %20, %3A, etc., with the actual character
  path = path:gsub("%%(%x%x)", function(hex)
    return string.char(tonumber(hex, 16))
  end)
  return path
end
```

**Reply by estin** [reply]
@habahabahaba 
Url object also has attribute `file_path - decodes the path field and interprets it as a file path`

**Reply by habahabahaba** [reply]
That simplifies things, thank you again.

**Comment by idr4n** [comment, marked as answer]
Actually, I have resolved this for me using the shell integration instead, that is, setting the window title via the shell. In my case, using fish, I accomplished what I wanted (and more, e.g., adding the git branch as well), with the following:

```fish
function fish_title
    # Get shortened path without the full home path
    # basename (prompt_pwd)

    # Get current working directory
    set -l wd (prompt_pwd)

    # Try to get git branch
    set -l git_branch (command git rev-parse --abbrev-ref HEAD 2>/dev/null)

    if test -n "$git_branch"
        echo "$wd ($git_branch)"
    else
        echo $wd
    end
end
```
